model/distro/distro.go

```diff
@@ -119,6 +119,34 @@ func (d *Distro) IsParent(s *evergreen.Settings) bool {
 	return false
 }
 
+func (d *Distro) GetImageID() string {
+	switch d.Provider {
+	case evergreen.ProviderNameEc2Auto:
+		s, _ := (*d.ProviderSettings)["ami"].(string)
+		return s
+	case evergreen.ProviderNameEc2OnDemand:
+		s, _ := (*d.ProviderSettings)["ami"].(string)
+		return s
+	case evergreen.ProviderNameEc2Spot:
+		s, _ := (*d.ProviderSettings)["ami"].(string)
+		return s
+	case evergreen.ProviderNameDocker:
+		s, _ := (*d.ProviderSettings)["image_url"].(string)
+		return s
+	case evergreen.ProviderNameDockerMock:
+		s, _ := (*d.ProviderSettings)["image_url"].(string)
+		return s
+	case evergreen.ProviderNameGce:
+		s, _ := (*d.ProviderSettings)["image_name"].(string)
+		return s
+	case evergreen.ProviderNameVsphere:
+		s, _ := (*d.ProviderSettings)["template"].(string)
+		return s
```

Comment:
we shouldn't ignore the ok check.

Reply:
If ok is false should we return an empty string or an error?